### PR TITLE
Add support for ImageComponent prop

### PR DESCRIPTION
**Pull request by jsamr**
- `ImageComponent` prop, implementation and documentation
- `onRemoveImage` added to documentation

### CODE REVIEW

File: src/CNRichTextEditor.js [hunk 1]
```diff
@@ -450,7 +450,7 @@ class CNRichTextEditor extends Component {
         const { width, height } = image.size;
         let myHeight = (this.state.layoutWidth - 4 < width) ? height * ((this.state.layoutWidth - 4) / width) : height; 
         let myWidth = (this.state.layoutWidth - 4 < width) ? this.state.layoutWidth - 4 : width;
-        
+        const { ImageComponent = Image } = this.props
```
Comment (ohpax) [inline src/CNRichTextEditor.js]: does this mean if they pass both we will only use the Image one?
I think that makes sense.

Comment (jsamr) [reply src/CNRichTextEditor.js]: If the library consumer doesn't give an `ImageComponent` prop, the default value `Image` will be used instead. That's the [destructuring assignment + default value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Default_values_2) :-)

Comment (ohpax) [reply src/CNRichTextEditor.js]: duh, 😄I don't know what I thought the first time when I saw this, cool.

Comment (jsamr) [reply src/CNRichTextEditor.js]: No worries :slightly_smiling_face:

**Pull request merged**